modules/components/src/Aggs/DatesAgg.js

```diff
@@ -16,55 +16,61 @@ const momentToBucketDate = moment => moment?.format(BUCKET_DATE_FORMAT);
-      focusedInput: null, // must be set with this.setInputFocus to bind with dom state
-    };
+    this.state = this.initializeState(props);
   }
 
   componentWillReceiveProps(nextProps) {
-    this.setState(this.getInitialRange(nextProps));
+    this.setState(this.initializeState(nextProps));
   }
 
-  getInitialRange = ({ buckets = [], getActiveValue = () => ({}) }) => {
+  initializeState = ({ buckets = [], getActiveValue = () => ({}) }) => {
     const { field } = this.props;
     const bucketMoments = buckets.map(x => bucketDateToMoment(x.key_as_string));
-    const minDate = minBy(bucketMoments, moment => moment.valueOf());
-    const maxDate = maxBy(bucketMoments, moment => moment.valueOf());
+    const minDate = minBy(bucketMoments, x => x.valueOf()).subtract(1, 'days');
+    const maxDate = maxBy(bucketMoments, x => x.valueOf()).add(1, 'days');
     const startFromSqon = getActiveValue({ op: '>=', field });
     const endFromSqon = getActiveValue({ op: '<=', field });
     return {
-      startDate: startFromSqon ? bucketDateToMoment(startFromSqon) : minDate,
-      endDate: endFromSqon ? bucketDateToMoment(endFromSqon) : maxDate,
+      minDate,
+      maxDate,
+      startDate: startFromSqon ? bucketDateToMoment(startFromSqon) : null,
+      endDate: endFromSqon ? bucketDateToMoment(endFromSqon) : null,
     };
   };
 
-  updateSqon = ({ startDate, endDate } = {}) => {
+  updateSqon = () => {
+    const { startDate, endDate } = this.state;
     const { field, handleDateChange } = this.props;
-    if (handleDateChange && field && startDate && endDate) {
+    if (handleDateChange && field) {
+      const content = [
+        ...(startDate
+          ? [
+              {
+                op: '>=',
+                content: {
+                  field,
+                  value: momentToBucketDate(startDate.startOf('day')),
+                },
+              },
+            ]
+          : []),
+        ...(endDate
+          ? [
+              {
+                op: '<=',
+                content: {
+                  field,
+                  value: momentToBucketDate(endDate.endOf('day')),
+                },
+              },
+            ]
+          : []),
+      ];
       handleDateChange({
         generateNextSQON: sqon =>
           replaceSQON(
-            {
-              op: 'and',
-              content: [
-                {
-                  op: '>=',
-                  content: {
-                    field,
-                    value: momentToBucketDate(startDate.startOf('day')),
-                  },
-                },
-                {
-                  op: '<=',
-                  content: {
-                    field,
-                    value: momentToBucketDate(endDate.endOf('day')),
-                  },
-                },
-              ],
-            },
-            sqon,
+            content.length ? { op: 'and', content } : null,
+            removeSQON(field, sqon),
```

Comment:
Just so I understand it fully, is this to handle the change above that's checking each field separately? :)

Reply:
yeah, basically remove both >= and <=, then add them in one-by-one